libs/platform/ebpf_epoch.c

```diff
@@ -248,45 +237,81 @@ ebpf_epoch_terminate()
 ebpf_result_t
 ebpf_epoch_enter()
 {
+    ebpf_result_t return_value;
     uint32_t current_cpu;
+    ebpf_epoch_state_t* epoch_state = NULL;
     current_cpu = ebpf_get_current_cpu();
+
     if (current_cpu >= _ebpf_epoch_cpu_count) {
         return EBPF_OPERATION_NOT_SUPPORTED;
     }
 
+    ebpf_lock_state_t state = ebpf_lock_lock(&_ebpf_epoch_cpu_table[current_cpu].lock);
+
     if (ebpf_is_preemptible()) {
-        return _ebpf_epoch_update_thread_state(current_cpu, ebpf_get_current_thread_id(), _ebpf_current_epoch, true);
+        ebpf_epoch_thread_entry_t* thread_entry =
+            _ebpf_epoch_get_thread_entry(current_cpu, ebpf_get_current_thread_id(), true);
```

Comment:
nit: to make this more readable can we use a comment describing the last bool param or use a constant identifier instead?